### PR TITLE
Improve search history

### PR DESCRIPTION
**Pull request by nicolaskempf57**
Close https://github.com/etalab/data.gouv.fr/issues/905

### CODE REVIEW

**Review by maudetes — COMMENTED** [review]
Yay! Seems to work as expected! Don't forget the changelog ;)

When going back, the search is not started again, could we re-search?

**Comment by maudetes** [comment]
> This should be working, do you have a js error when doing it ? I don't reproduce it yet.

[Capture d’écran vidéo de 21-10-2022 15:28:58.webm](https://user-images.githubusercontent.com/28541902/197207867-b1128bfd-c051-4d95-a4d2-97970c8a033e.webm)

I don't have any error in the console. I reproduce it on Firefox and Chrome.

**Comment by nicolaskempf57** [comment]
> I don't have any error in the console. I reproduce it on Firefox and Chrome.

I tried it by deleting text in the input and not with the back button, now I reproduce it ! 

This is a bug already existing in production, do we have any issue to link to ? I'll fix it but I have to figure out how to update the search parameters too.

File: theme/js/components/search/search.vue [hunk 1]
```diff
@@ -297,18 +296,36 @@ export default defineComponent({
...
     };
 
+    const SAVE_TO_HISTORY = true;
+    const DONT_SAVE_TO_HISTORY = false;
+
+    const updateUrl = (save = SAVE_TO_HISTORY) => {
+      // Update URL to match current search params value for deep linking
+      let url = new URL(window.location.href);
+      url.search = new URLSearchParams(searchParameters.value).toString();
+      console.log(save);
```
Comment (maudetes) [inline theme/js/components/search/search.vue:306]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 - Switch from `Flask-restplus` to its fork `Flask-rest-x` [#165](https://github.com/etalab/udata-front/pull/165)
 - Add CaptchEtat integration [#159](https://github.com/etalab/udata-front/pull/159)
   - new environment variables : CAPTCHETAT_BASE_URL, CAPTCHETAT_OAUTH_BASE_URL, CAPTCHETAT_CLIENT_ID and CAPTCHETAT_CLIENT_SECRET
+- Improve search history [#162](https://github.com/etalab/udata-front/pull/162)
```
Comment (maudetes) [inline CHANGELOG.md:8]: You will need to move this entry to the new current section.

**Review by maudetes — APPROVED** [review]
Works perfectly!

File: CHANGELOG.md (hunk 2 above)
Comment (nicolaskempf57) [reply CHANGELOG.md]: Done :)

**Pull request merged**